microbench/src/main/java/io/netty/microbench/headers/HeadersBenchmark.java

```diff
@@ -139,4 +163,546 @@ public void http2Iterate(Blackhole bh) {
             bh.consume(entry);
         }
     }
+
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    public void httpAddAllFastest(Blackhole bh) {
+        bh.consume(new DefaultHttpHeaders(false).add(httpHeaders));
```

Comment:
So the law of unintended consequences kicked in....

There was a memory leak in DefaultHttp2Headers.clear as it didn't reset the pseudoHeader pointer to head and retained a reference to the data from the last run. This caused the benchmarks to go haywire as you might imagine. Once I fixed that the numbers were about equivalent to what I got before.

```
Benchmark                            (exampleHeader)  Mode  Cnt     Score     Error  Units
HeadersBenchmark.http2AddAllFastest             FIVE  avgt   10   136.940 ±   3.899  ns/op
HeadersBenchmark.http2AddAllFast                FIVE  avgt   10   199.410 ±   3.022  ns/op
HeadersBenchmark.http2AddAllSlow                FIVE  avgt   10   231.186 ±   8.891  ns/op
HeadersBenchmark.http2AddAllFastest        TWENTYTWO  avgt   10   479.186 ±   6.863  ns/op
HeadersBenchmark.http2AddAllFast           TWENTYTWO  avgt   10  1014.660 ±  13.290  ns/op
HeadersBenchmark.http2AddAllSlow           TWENTYTWO  avgt   10   971.376 ±  12.329  ns/op
HeadersBenchmark.httpAddAllFastest              FIVE  avgt   10   100.142 ±   2.534  ns/op
HeadersBenchmark.httpAddAllFast                 FIVE  avgt   10   244.474 ±   3.014  ns/op
HeadersBenchmark.httpAddAllSlow                 FIVE  avgt   10   233.950 ±  33.473  ns/op
HeadersBenchmark.httpAddAllFastest         TWENTYTWO  avgt   10   415.668 ±  24.973  ns/op
HeadersBenchmark.httpAddAllFast            TWENTYTWO  avgt   10  1743.317 ±  13.812  ns/op
HeadersBenchmark.httpAddAllSlow            TWENTYTWO  avgt   10  1673.307 ± 125.075  ns/op
```